UnityProject/Assets/Scripts/Disposals/DisposalIntake.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Disposals
+{
+	public class DisposalIntake : DisposalMachine, IServerDespawn, IExaminable
+	{
+		const float ANIMATION_TIME = 1.2f; // As per sprite sheet JSON file.
+		const float FLUSH_DELAY = 1;
+
+		DirectionalPassable directionalPassable;
+		DisposalVirtualContainer virtualContainer;
+
+		public bool IsOperating { get; private set; }
+
+		private enum SpriteState
+		{
+			Idle = 0,
+			Operating = 1
+		}
+
+		#region Lifecycle
+
+		protected override void Awake()
+		{
+			base.Awake();
+
+			if (TryGetComponent(out Directional directional))
+			{
+				directional.OnDirectionChange.AddListener(OnDirectionChanged);
+			}
+			directionalPassable = GetComponent<DirectionalPassable>();
+			DenyEntry();
+		}
+
+		void Start()
+		{
+			UpdateSpriteOrientation();
+		}
+
+		public void OnDespawnServer(DespawnInfo info)
+		{
+			if (virtualContainer != null) Despawn.ServerSingle(virtualContainer.gameObject);
+		}
+
+		#endregion Lifecycle
+
+		// Woman! I can hardly express
+		// My mixed motions at my thoughtlessness
+		// TODO: Don't poll, find some sort of trigger for when an entity enters the same tile.
+		void UpdateMe()
+		{
+			if (!MachineSecured || IsOperating) return;
+			GatherEntities();
+		}
+
+		private void OnDirectionChanged(Orientation newDir)
+		{
+			UpdateSpriteOrientation();
+		}
+
+		void SetIntakeOperating(bool isOperating)
+		{
+			IsOperating = isOperating;
+			UpdateSpriteState();
+		}
+
+		#region Sprites
+
+		void UpdateSpriteState()
+		{
+			if (IsOperating)
+			{
+				baseSpriteHandler.ChangeSprite((int) SpriteState.Operating);
+			}
+			else
+			{
+				baseSpriteHandler.ChangeSprite((int) SpriteState.Idle);
+			}
+		}
+
+		void UpdateSpriteOrientation()
+		{
+			switch (directionalPassable.Directional.CurrentDirection.AsEnum())
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)